lib/demangler/llvm.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import {CompilationResult} from '../../types/compilation/compilation.interfaces';
+import {LLVMOptPipelineOutput} from '../../types/compilation/llvm-opt-pipeline-output.interfaces';
+import {ResultLine} from '../../types/resultline/resultline.interfaces';
+
+import {logger} from '../logger';
+import {SymbolStore} from '../symbol-store';
+import * as utils from '../utils';
+
+import {BaseDemangler} from './base';
+import {PrefixTree} from './prefix-tree';
+
+export class LLVMIRDemangler extends BaseDemangler {
+    llvmSymbolRE = /@([\w$.]+)/gi;
+
+    static get key() {
+        return 'llvm-ir';
+    }
+
+    override collectLabels() {
+        for (const line of this.result.asm) {
+            const text = line.text;
+            if (!text) continue;
+
+            const matches = [...text.matchAll(this.llvmSymbolRE)];
+            for (const match of matches) {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                this.symbolstore!.add(match[1]);
+            }
+        }
+    }
+
+    collect(result: {asm: ResultLine[]}) {
+        this.result = result;
+        if (!this.symbolstore) {
+            this.symbolstore = new SymbolStore();
```

Comment:
since you only create this once anyway per compilation, the constructor is a more logical place for this